Use docker-ce, configure with daemon.json file

manics opened the pull request:
This is a breaking change (tag `2.0.0`), see `CHANGES.md`.

Changes:
- Use Docker Community Edition in accordance with Docker's changed releases.
- Use `daemon.json` for configuring docker. This means it's no longer necessary to override the systemd file, we can use the unmodified default version.
- Adds a molecule test
- Adds an install-only option (disable systemd)

manics committed:
Refactor docker to use daemon.json file

Custom network/storage is still to be tested

Also adds molecule tests.

manics committed:
Update travis to run molecule test

manics committed:
Fix format of daemon.json storage opts

manics committed:
Add experimental and debug options

manics committed:
docker-engine is now docker-ce (community edition)

manics committed:
Update README.md, add breaking CHANGES.md

joshmoore commented:
Have you tested with experimental? Could that be an additional test? Any other similar parameters (esp. ones that we are unlikely to be using)?

Otherwise, :+1: for getting this in as `2.0.0`.

manics committed:
Remove docker_experimental

manics committed:
Rearrange readme, add another example playbook.

manics commented:
It's probably inadvisable to use `experimental` in production so I've taken it out. I've also rearranged the readme so the most useful variables are first.

manics committed:
Allow storage-driver to be overridden

This is mainly useful for testing Docker or other roles when a full VM is not available

manics committed:
Test that `docker run` works

Use VFS driver when testing on travis (docker-in-docker)

manics commented:
Last three commits switch the molecule test to use the docker VFS driver when testing in docker (which involves Docker-in-Docker), since the default filesystem may not work.

manics committed:
Add docker_systemd_setup option to install but not enable docker

manics committed:
Test docker_systemd_setup=False

manics commented:
And finally I've added commits to allow installation of Docker without enabling/running it. This means you can use the role to setup a virtual image without needing systemd or other required services to be running (e.g. for devspace @aleksandra-tarkowska)

atarkowska commented:
can confirm it works

joshmoore committed:
Fix typo

joshmoore commented on molecule.yml:
```diff
@@ -0,0 +1,50 @@
+docker:
+  containers:
+  - name: docker
+    image: manics/centos-systemd-ip-docker
```
Why not https://hub.docker.com/r/openmicroscopy/centos-systemd-ip/ ?

manics replied:
Most likely because I copied it from some other out-of-date file, now updated.

manics committed:
Use openmicroscopy/centos-systemd-ip

joshmoore commented:
:+1: merging and tagging.
